### PR TITLE
Add lint-test JS Github action

### LINKED ISSUE #32311: Add JS lint/test Github action (post WooCommerce Admin merge)
### Describe the solution you'd like

We have a ["lint and test JS"](https://github.com/woocommerce/woocommerce-admin/actions/workflows/lint-test-js.yml) Github action in the woocommerce-admin to test and lint js code.

But we don't have an existing Github workflow in the monorepo. Only found [one](https://github.com/woocommerce/woocommerce/blob/trunk/.github/workflows/package-js-api.yml) that tests woocommerce/api JS package.

We should add an action (or actions) to test and lint our code.

Use `paths` to only run the action when js packages and `./client` are changed.

```
on:
  pull_request:
    paths:
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chihsuan**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #32311.

Add lint-test-js Github action

- Update "test" command and exclude api-core-tests to pass tests (`api-core-tests` tests are not pure unit tests. It requires a WP instance.)

### How to test the changes in this Pull Request:

- [This action](https://github.com/woocommerce/woocommerce/runs/5718848306?check_suite_focus=true) should pass on the test branch
- Run the action commands on your local

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

Dev: Add lint-test-js Github action

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: .github/workflows/pr-lint-test-js.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        with:
+          node-version: '16'
+      
+      - name: Install PNPM and install dependencies
+        run: npm install -g pnpm && pnpm install
+
+      - name: Lint
+        run: |
+          pnpm nx lint woocommerce-admin
+          pnpm nx lint:js:packages woocommerce-admin
```
Comment (louwie17) [inline .github/workflows/pr-lint-test-js.yml]: This command doesn't seem to work for me locally, but I notice it works fine on Github. Do we maybe want to change it to `lint:js-packages`?
I did notice @rjchow comment in Slack ( p1648451119236319-slack-CBKK5KM41 ), but the `run-many` work around didn't work for me either.

File: .github/workflows/pr-lint-test-js.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+on:
+  pull_request:
+    paths:
+      - 'packages/js/**/**'
+      - 'plugins/woocommerce-admin/client/**'
+      - '!**.md'
+
+jobs:
+  lint-test-js:
+    runs-on: ubuntu-18.04
```
Comment (louwie17) [inline .github/workflows/pr-lint-test-js.yml:13]: Maybe add a `name: Lint and Test JS` so that it doesn't show as `lint-test-js` within Github

**Review by louwie17 — COMMENTED** [review]
Thanks for adding this action and removing the old one's 💯 
This for the most part tested well for me, I just wasn't able to run the `lint:js:packages` command as mentioned in a comment below.
Also left a suggestion to add a `name` to the actual. job.

File: .github/workflows/pr-lint-test-js.yml (hunk 1 above)
Comment (chihsuan) [reply .github/workflows/pr-lint-test-js.yml]: > This command doesn't seem to work for me locally, but I notice it works fine on Github. Do we maybe want to change it to lint:js-packages?

This is interesting. Both commands can be run in my local. But yes, I think we can change it to `lint:js-packages` to make it work for everyone.

File: .github/workflows/pr-lint-test-js.yml (hunk 2 above)
Comment (chihsuan) [reply .github/workflows/pr-lint-test-js.yml:13]: Sounds good! Added in 71ba4c748a8267e34e9589c809b1bc7126a6fd00.

File: .github/workflows/pr-lint-test-js.yml (hunk 1 above)
Comment (chihsuan) [reply .github/workflows/pr-lint-test-js.yml]: Changed in a2327bc90338965ae84689c2dbd7c7a7ed3c7391.

**Review by louwie17 — APPROVED** [review]
Thanks for the updates @chihsuan, things are running fine on my end now. I am not sure why it runs for you and doesn't for me (unless I somehow got an older `nx` version). Anyway this fixes it for sure.
This is good to merge 🚀

**Pull request merged**